superset/commands/explore/permalink/create.py

```diff
@@ -65,15 +66,13 @@ def run(self) -> str:
             "datasource": self.datasource,
             "state": self.state,
         }
-        command = CreateKeyValueCommand(
-            resource=self.resource,
-            value=value,
-            codec=self.codec,
-        )
-        key = command.run()
-        if key.id is None:
+        entry = KeyValueDAO.create_entry(self.resource, value, self.codec)
+        db.session.flush()
+        key = entry.id
+        if key is None:
             raise ExplorePermalinkCreateFailedError("Unexpected missing key id")
-        return encode_permalink_key(key=key.id, salt=self.salt)
+        db.session.commit()  # pylint: disable=consider-using-transaction
```

Comment:
Why do you need `commit` here given that the method is annotated with `@transaction`?

Reply:
Whoops, that wasn't meant to be there